src/common.c

```diff
@@ -1082,7 +1083,17 @@ ssize_t ofi_bsock_send(struct ofi_bsock *bsock, const void *buf, size_t *len)
 	}
 
 	assert(!ofi_bsock_tosend(bsock));
-	ret = ofi_send_socket(bsock->sock, buf, bytes, MSG_NOSIGNAL);
+	if (bytes > bsock->zerocopy_size) {
+		ret = ofi_send_socket(bsock->sock, buf, bytes,
+				      MSG_NOSIGNAL | OFI_ZEROCOPY);
+		if (ret >= 0) {
+			bsock->async_index++;
+			*len = ret;
```

Comment:
Shouldn't we change the API of ofi_bsock_send() and always return the number of bytes transferred (or waiting for transmission) in the in/out argument `len`, even if zerocopy is not used?

Reply:
Agree - will update as it makes the API easier to use.  I had it this way to avoid the extra write, but the call sites end up having to update anyway.